### PR TITLE
Hand of god gamemode

### PR DESCRIPTION
**Pull request by airlines7**
# Document the changes in your pull request
## main things
Adds the new gamemode Hand of God
The cults of Nar'sie and Ratvar face each other to summon their gods with the crew in the middle.

## Cool, anything special?
yes, clockcult in this mode does not use reebe and remains on the station, they become more like bloodcult in this regard, expected to make a "city of cogs" in maints or wherever on the station. They also get a silly point defense phase like bloodcult.
- 6 silly antags roundstart, 3 bloodcult, 3 clockcult.

## How a round should go: bloodcult

- starts out like a bloodcult round, just that there are two cults.
-  At this time they shouldn't be aware of eachother.
- bloodcult is like normal, they convert people and act all cult like. 
- midround cult should be larger and in charger, they should know of clockcult by this point and fights between the two begin.
- after obtaining halos, they should have the ability to convert clockcult directly with 3 people.
- time for point defense, bloodpillar style.
- nar'sie is summoned.

## How a round should go: clockcult

- coggers are just like bloodcult roundstart, they convert people and act all cult like.
- midround is where the difference begins, with the conversions of more people OR the building of cogscarabs (limited to an undecided amount). Just like bloodcult after a certain amount they gain their golden light and can convert bloodcult directly with a longer conversion time on the ominous sigil.
- With a certain amount of servants + cog scarabs + enough energy, they have the minimum to construct the cogpillars (placeholder name) 
basically 4 bloodpillars but cog flavored, once constructed they alter the landscape. Main difference is clockcult knows where they need to build the pillars round start, and have the advantage of building typical cogger traps, machines, etc.
- all cogpillars are constructed point defense triggers as the pillars activate, time for point defense, cogpillar style.
- ratvar is summoned.

# Spriting
WIP - Cogpillars 

# Wiki Documentation

cogpillars, why this version of clockcult exists, HoG gamemode, etc.
There is gonna be a lot, once it's all ironed out I'll go over this if I remember.

# Wow cool, why are you even doing this?
Main reason is at least one person on yog wanted this gamemode (me). There never is any interaction between the two cults even though lore wise they are at a war (I think? Lore is always changing). So I hope that this gamemode changes that. I think there is a lot of silly RP to do with this gamemode, and even without RP I believe the gamemode will be a good large scale gamemode to have in the rotation or as just an event gamemode.

Anyways this is a draft, coding is slow but I'm making progress. If you want to help feel free to.

At the moment of uploading this, the basic clockcult is setup for the antag path, thats about it. I just want some more eyes on the code as I go.

# Changelog

:cl:  
experimental: hand of god
/:cl:

### CODE REVIEW

**Comment by iamthesilly** [comment]
Wait so, bloodcult can convert clock cult but not the other way around? That seems one-sided.

**Comment by airlines7** [comment]
> Wait so, bloodcult can convert clock cult but not the other way around? That seems one-sided.

ah no I just forgot to type that, let me add that.

**Comment by dcxgame** [comment]
Does clock cult spawn on the station or in reebe in this? If they spawn on station, and aren't supposed to know of each other, doesn't this present an impossible request for the clock cult players to pretend they don't know the gamemode, based on where they started?

**Comment by Hislyna** [comment]
Should probably be an admin only mode instead of one in rotation. Thats my take tho.

**Comment by LaOubliette** [comment]
sounds fun as hell, i've wanted to see a conversion antag vs conversion antag vs crew since i found out about Capitalist/Communist golems

**Comment by neeshacark** [comment]
Ah - here we go again.  Will it actually get finished this time? (And not closed by bibby)

File: code/game/gamemodes/hand_of_god/hand_of_god.dm [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+			return FALSE
+	else
+		return FALSE
+	if(HAS_TRAIT(M, TRAIT_MINDSHIELD) || issilicon(M) || isbot(M) || isdrone(M) || ismouse(M) || is_servant_of_ratvar(M) || !M.client)
+		return FALSE //can't convert machines, shielded, braindead, mice, or ratvar's dogs
+	return TRUE
+*/
+/datum/game_mode
+	var/list/hog_clockcult = list() //The Enlightened servants of Ratvar
+	var/list/hog_cult = list() //The devoted cult of Nar'sie
```
Comment (JohnFulpWillard) [inline code/game/gamemodes/hand_of_god/hand_of_god.dm:117]: Do not do this. members should be stored on the mind and not the gamemode. See https://github.com/yogstation13/Yogstation/pull/19524 for better information on it

File: code/game/gamemodes/hand_of_god/hand_of_god.dm [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+	return ..()
+
+/datum/game_mode/hand_of_god/check_finished(force_ending)
+	if (..())
+		return TRUE
+	return !hand_of_narsie.check_sacrifice_status() //we should remove this any time soon
+
+//clockcult
+
+	for(var/datum/mind/servant)
```
Comment (JohnFulpWillard) [inline code/game/gamemodes/hand_of_god/hand_of_god.dm:208]: This for loops over what?

**Review by JohnFulpWillard — COMMENTED** [review]
Didn't read through most of it because I don't actually know what this antagonist is supposed to be. Is this just blood cult and clock cult spawning in the same round?
You say there's differences between clockies in the two modes. Won't this make it easily metagamed (essentially telling clock cultists that there are blood cultists), and confusing for players who now need to be able to tell two completely separate antags apart despite having a majority of features be the exact same?
Would it not be better to have both antagonists the same in both modes, the only difference being that this one will spawn both? Interactions between the two cultists can be generalized so admins can spawn cultists without having to run the snowflake gamemode.

**Pull request closed without merging**